### ISSUE TITLE
Gastoken countermeasures due to rapidly growing blockchain size

### ISSUE THREAD

**Issue by TheEnthusiasticAs** [ethereumclassic/ECIPs#253]
A place to discuss possible countermeasures against rapidly growing ETC blockchain size. 
One of the solutions was already proposed here #14 by Anthony back in january 2019.

**Comment by phyro** [comment]
relevant description of refund issue https://ethereum.corepaper.org/state/refund/
The above link also has a link to the discussion https://github.com/corepaper/ethereum/issues/4

**Comment by iquidus** [comment]
Unfeasible blockchain growth is an unfortunate side-effect of properties inherited from upstream ETH. This was one of the things we considered when launching Ubiq with slower target times for blocks. Whiteblock did a great analysis of our parameters vs upstream ETH at various block gas limits here:

* case study: https://www.whiteblock.io/wp-content/uploads/2019/07/ubiq-case-study.pdf 
* report: https://whiteblock.io/wp-content/uploads/2019/07/ubiq-report.pdf

Well worth the read if considering making changes to layer 1. This approach doesn't sacrifice contract size/complexity while still achieving the goal of slowing down chain growth.

Some knock-on affects to consider however: 

* ETHs difficulty algorithm performs poorly with slower blocktimes, we designed Flux to address this.
* DAG size is based on epoch, epoch is defined in blocks (30k), longer blocks mean longer epoch (in terms of time). DAG grows slower. In our case still allowing support for 2GB cards.
* Monetary policy is defined in blocks, this would need to be adjusted to maintain current schedule.
* Less uncle blocks (and reduced inflation from uncle block rewards)
* Uncles currently get rewarded up to a depth of 7 blocks old, this doesn't make sense with slower blocks (e.g at 90seconds 7 blocks > 10minutes). We reduced ours to a depth of 1 block old only (see here for more details: https://bitcointalk.org/index.php?topic=1631210.msg17154836#msg17154836)

This is all doable on ETC of course, but it is fairly intrusive and touches upon multiple areas of consensus (target time/diff algo, monetary policy), requiring a hard fork (or forks).

**Comment by realcodywburns** [comment]
#432 to be solved as part of ECIP 1104 hard fork

**Comment by realcodywburns** [comment]
moving to closed in 7 days

**Comment by erkinalp** [comment]
Still relevant due to increased hashpower caused by ETH having switched away from proof of work

**Comment by gitr0n1n** [comment]
> #432 to be solved as part of ECIP 1104 hard fork

Closing this discussion due to this resolution. Looks like we may want to close ECIP-1047 as well. https://github.com/ethereumclassic/ECIPs/issues/14

We did observe mining pools accidentally adjust their client gas limit and this restricted some smart contract from deploying. This was corrected very quickly, but showed the effects of lowering the gas and the negative consequences to the development ecosystem on Classic.
